### PR TITLE
Add memoized_only mode to tensor conversion

### PR DESCRIPTION
**Pull request by voznesenskym**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #99789
* #99742
* __->__ #99741
* #99566

### CODE REVIEW

File: torch/_subclasses/fake_tensor.py [hunk 1]
```diff
@@ -253,10 +253,13 @@ def from_real_tensor(
         source=None,
         dynamic_dims: Optional[DimList[DimDynamic]] = None,
         constraint_dims: Optional[DimList[DimConstraint]] = None,
+        memoized_only=False,
     ):
         maybe_memo = self._get_memo(t)
         if maybe_memo is not None:
             return maybe_memo
+        if memoized_only:
+            return None
```
Comment (ezyang) [inline torch/_subclasses/fake_tensor.py:262]: Better to raise an exception?

**Review by ezyang — APPROVED** [review]

File: torch/_subclasses/fake_tensor.py (hunk 1 above)
Comment (voznesenskym) [reply torch/_subclasses/fake_tensor.py:262]: I prefer None here. You asked for it via API, nothing is exceptional.

**Comment by malfet** [comment]
This broke MacOS x86_64 OSS builds, but to be frank, they are quite ridiculous, i.e. building for MacOS 10.9, which was released back in 2013... PR updating build rules is coming

**Pull request closed without merging**